core-blocks/embed/index.js

```diff
@@ -87,44 +82,88 @@ function getEmbedBlockSettings( { title, description, icon, category = 'embed',
 			}
 		},
 
-		edit: class extends Component {
+		edit: compose(
+			withSelect( ( select, ownProps ) => {
+				const { url } = ownProps.attributes;
+				// Preview is undefined if we don't know the status of it, false if it failed,
+				// otherwise it will be an object containing the embed response.
+				const preview = url ? select( 'core' ).getEmbedPreview( url ) : undefined;
+				return {
+					preview,
+				};
+			} )
+		)( class extends Component {
 			constructor() {
 				super( ...arguments );
 
-				this.doServerSideRender = this.doServerSideRender.bind( this );
+				this.setUrl = this.setUrl.bind( this );
+				this.processPreview = this.processPreview.bind( this );
 
 				this.state = {
 					html: '',
 					type: '',
 					error: false,
 					fetching: false,
 					providerName: '',
+					url: '',
 				};
 			}
 
-			componentDidMount() {
-				this.doServerSideRender();
+			componentWillMount() {
+				if ( this.props.attributes.url ) {
+					// Loading from a saved block, set the state up and display the fetching UI.
+					this.setState( { fetching: true, url: this.props.attributes.url } );
+					if ( this.props.preview ) {
+						// Preview must have already been fetched prior to loading this block, so process it.
+						this.processPreview( this.props.preview, this.props.attributes.url );
+					}
+				}
 			}
 
 			componentWillUnmount() {
 				// can't abort the fetch promise, so let it know we will unmount
 				this.unmounting = true;
 			}
 
+			componentWillReceiveProps( nextProps ) {
```

Comment:
`componentWillReceiveProps` is a deprecated lifecycle function. For your purpose here, you might consider `componentDidUpdate`, though it will incur a secondary render unfortunately. `getDerivedStateFromProps` is another option, though will require more substantial refactor.

https://reactjs.org/docs/react-component.html#the-component-lifecycle